app/src/main/java/de/westnordost/streetcomplete/quests/building_type/BuildingTypeItem.kt

```diff
@@ -0,0 +1,76 @@
+package de.westnordost.streetcomplete.quests.building_type
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.view.image_select.Item
+import de.westnordost.streetcomplete.quests.building_type.BuildingType.*
+
+fun List<BuildingType>.toItems() = this.map { it.asItem() }
+
+fun BuildingType.asItem(): Item<String> = when(this) {
```

Comment:
Here, it should be 

`fun BuildingType.asItem(): Item<BuildingType> = when(this) {`

not string

Reply:
When it returns a `Item<BuildingType>`, the logic in [`AGroupedImageListQuestAnswerFragment.kt`](https://github.com/streetcomplete/StreetComplete/blob/9820af988819cae4e0ee33a933c5b53cfc29919f/app/src/main/java/de/westnordost/streetcomplete/quests/AGroupedImageListQuestAnswerFragment.kt#L99-L128) (don't accept a `null` answer) won't work anymore, because the answer is then a non-null Item instead of a null String:

https://github.com/streetcomplete/StreetComplete/blob/9820af988819cae4e0ee33a933c5b53cfc29919f/app/src/main/java/de/westnordost/streetcomplete/quests/AGroupedImageListQuestAnswerFragment.kt#L99-L128

Reply:
Except if `FOR_CARS` would not actually be a BuildingType but just a categorization for the UI. Which is kinda true.